client/packages/programs/src/JsonForms/components/Search/SearchWithDocumentSource.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useEffect } from 'react';
+import { ControlProps } from '@jsonforms/core';
+import { get as extractProperty } from 'lodash';
+import {
+  useTranslation,
+  Box,
+  FormLabel,
+  Typography,
+} from '@openmsupply-client/common';
+import {
+  FORM_LABEL_COLUMN_WIDTH,
+  FORM_INPUT_COLUMN_WIDTH,
+} from '../../common/styleConstants';
+import { z } from 'zod';
+import { useZodOptionsValidation } from '../../common/hooks/useZodOptionsValidation';
+import { usePatientStore } from 'packages/programs/src/hooks';
+import { useEncounter } from '../../../api';
+import { RegexUtils, JSXFormatters } from '@openmsupply-client/common';
+
+const Options = z
+  .object({
+    source: z.enum(['document']),
+    /**
+     * Which pre-defined query to use (in useSearchQueries)
```

Comment:
```suggestion
     * Source document type. Either the current patient or the current encounter (if applicable).
```

May `current` instead of `encounter` because you might not be in an encounter. Than it would also work for program enrollments